docs: agents tutorial (#8961)
langflow-ai/langflow: docs/docs/Tutorials/agent.md

---
title: Connect applications to agents
slug: /agent-tutorial
---

import Icon from "@site/src/components/icon";
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

This tutorial shows you how to connect a JavaScript application to a Langflow [agent](/agents).

With the agent connected, your application can use any connected tools to retrieve more contextual and timely data without changing any application code. The tools are selected by the agent's internal LLM to solve problems and answer questions.

## Prerequisites

- [A running Langflow instance](/get-started-installation)
- [A Langflow API key](/configuration-api-keys)
- [An OpenAI API key](https://platform.openai.com/api-keys)
- [Langflow JavaScript client installed](/typescript-client)

This tutorial uses an OpenAI LLM. If you want to use a different provider, you need a valid credential for that provider.

## Create an agentic flow

> The following steps modify the [**Simple agent**](/simple-agent) template to connect [**Directory**](/components-data#directory) and [**Web search**](/components-data#web-search) components as tools for the agent.

Reviewer: Should we capitalize "Agent" here as it is in Langflow?

The Directory component loads all files of a given type from a target directory on your local machine, and the Web search component performs a DuckDuckGo search.

1. In Langflow, click **New Flow**, and then select the **Simple agent** template.
2. Remove the **URL** and **Calculator** tools, and drag the [**Directory**](/components-data#directory) and [**Web search**](/components-data#web-search) components to your workspace.
3. In the **Directory** component's **Path** field, enter the path to your directory, and the types of files you want to provide to your agent.
In this example, the directory name is `customer_orders` and the file type is `.csv`, because you want the agent to have access to a record of customer purchases.

> If you don't have a csv on hand, you can download [customer-orders.csv](/files/customer_orders/customer_orders.csv) and save it in a folder called `customer_orders`.

Reviewer: Maybe "save it in a folder locally..." ? Just to clarify.

6. To verify that your flow is operational, click <Icon name="Play" aria-hidden="true" /> **Playground**, and then ask the LLM a question, such as `Recommend 3 used items for carol.davis@example.com, based on previous orders.`
The LLM should respond with recommendations and web links for items based on previous orders in `customer_orders.csv`.
The Playground displays the agent's chain of thought as it uses the Directory component's `as_dataframe` tool to retrieve a [DataFrame](/concepts-objects#dataframe-object), and the Web search component's `perform_search` tool to find links to related items.

## Send requests to your flow from a JavaScript application

With your flow operational, connect it to a JavaScript application to use the agent's responses.

In this example, the application sends a customer's email address to the Langflow agent. The agent compares the customer's previous orders within the Directory component, searches the web for used versions of those items, and returns three results.

1. To include the email address as a value in your flow, add a [Prompt](/components-prompts) component to your flow between the **Chat Input** and **Agent**.

> 2. In the Prompt component's **Template** field, enter `Recommend 3 used items for ${email}, based on previous orders.`

Reviewer: I think it might be worth calling out that adding the curly brackets creates a variable in the "Prompt" component.

    The flow appears like this.

    ![An agent component connected to web search and directory components](/img/tutorial-agent-with-directory.png)

3. To construct a JavaScript application to connect to your flow, gather the following information:

    * `LANGFLOW_SERVER_ADDRESS`: Your Langflow server's domain. The default value is `127.0.0.1:7860`. You can get this value from the code snippets on your flow's [**API access** pane](/concepts-publish#api-pane).
    * `FLOW_ID`: Your flow's UUID or custom endpoint name. You can get this value from the code snippets on your flow's [**API access** pane](/concepts-publish#api-pane).
    * `LANGFLOW_API_KEY`: A valid Langflow API key. To create an API key, see [API keys](/configuration-api-keys).

2. Copy the following script into a JavaScript file, and then replace the placeholders with the information you gathered in the previous step.

> If you're using the `customer_orders.csv` example file, you can run this as-is with the example email address.

Reviewer: You mention about being able to run the flow "as-is" with the `customer_orders.csv` how about when the user wants to use a different file? Could you perhaps show an example for that case? Or is the assumption that the file directory is "hard-coded" in the LF flow?

---
title: Connect applications to agents
slug: /agent-tutorial
---

import Icon from "@site/src/components/icon";
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

This tutorial shows you how to connect a JavaScript application to a Langflow [agent](/agents).

With the agent connected, your application can use any connected tools to retrieve more contextual and timely data without changing any application code. The tools are selected by the agent's internal LLM to solve problems and answer questions.

## Prerequisites

- [A running Langflow instance](/get-started-installation)
- [A Langflow API key](/configuration-api-keys)
- [An OpenAI API key](https://platform.openai.com/api-keys)
- [Langflow JavaScript client installed](/typescript-client)

This tutorial uses an OpenAI LLM. If you want to use a different provider, you need a valid credential for that provider.

## Create an agentic flow

> The following steps modify the [**Simple agent**](/simple-agent) template to connect [**Directory**](/components-data#directory) and [**Web search**](/components-data#web-search) components as tools for the **Agent**.

Reviewer 2: ```suggestion
The following steps modify the [**Simple agent**](/simple-agent) template to connect [**Directory**](/components-data#directory) and [**Web search**](/components-data#web-search) components as tools for an **Agent** component.
```

> The Directory component loads all files of a given type from a target directory on your local machine, and the Web search component performs a DuckDuckGo search.

Author: ```suggestion
The **Directory** component loads all files of a given type from a target directory on your local machine, and the **Web search** component performs a DuckDuckGo search.
When connected to an **Agent** component as tools, the agent has the option to use these components when handling requests.
```

1. In Langflow, click **New Flow**, and then select the **Simple agent** template.

> 2. Remove the **URL** and **Calculator** tools, and drag the [**Directory**](/components-data#directory) and [**Web search**](/components-data#web-search) components to your workspace.

Reviewer 2: ```suggestion
2. Remove the **URL** and **Calculator** tools, and  then drag the [**Directory**](/components-data#directory) and [**Web search**](/components-data#web-search) components into your workspace.
```

5. For both tool components, connect the **Toolset** port to the Agent component's **Tools** port.
6. In the **Agent** component, enter your OpenAI API key.

    If you want to use a different provider or model, edit the **Model Provider**, **Model Name**, and **API Key** fields accordingly.

7. To test the flow, click <Icon name="Play" aria-hidden="true" /> **Playground**, and then ask the LLM a question, such as `Recommend 3 used items for carol.davis@example.com, based on previous orders.`

    Given the example prompt, the LLM would respond with recommendations and web links for items based on previous orders in `customer_orders.csv`.

    The **Playground** prints the agent's chain of thought as it selects tools to use and interacts with functionality provided by those tools.
    For example, the agent can use the **Directory** component's `as_dataframe` tool to retrieve a [DataFrame](/concepts-objects#dataframe-object), and the **Web search** component's `perform_search` tool to find links to related items.

## Add a prompt component to the flow

> With your flow operational, connect it to a JavaScript application to use the agent's responses.

Reviewer 2: Move this line down to the JS script section

In this example, the application sends a customer's email address to the Langflow agent. The agent compares the customer's previous orders within the Directory component, searches the web for used versions of those items, and returns three results.

1. To include the email address as a value in your flow, add a [Prompt](/components-prompts) component to your flow between the **Chat Input** and **Agent**.
2. In the Prompt component's **Template** field, enter `Recommend 3 used items for {email}, based on previous orders.`
Adding the `{email}` value in curly braces creates a new input in the **Prompt** component, and the component connected to the `{email}` port is supplying the value for that variable.
This creates a point for the user's email to enter the flow from your request.
If you aren't using the `customer_orders.csv` example file, modify the input to search for a value in your dataset.

>     The flow appears like this.

Reviewer 2: ```suggestion
    The flow now has <!-- list the components -->
```
